pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -456,8 +460,19 @@ func (c *Cluster) startOSDDaemonsOnPVC(pvcName string, config *provisionConfig,
 		return
 	}
 
+	if len(osdIDArray) == 0 {
+		osdIDArray, err = opconfig.GetOSDDeployments(c.context, c.Namespace, AppName)
+		if err != nil {
+			logger.Errorf("failed to fetch osd deployment %q, %v", AppName, err)
+		}
+	}
 	// start osds
 	for _, osd := range osds {
+		osdFound := opconfig.ExpandConditionCheck(osdIDArray, osd.ID)
+		if osdFound != true {
+			opconfig.ConditionExport(c.context, c.Namespace, c.clusterInfo.Name,
+				cephv1.ConditionExpanding, v1.ConditionTrue, "ClusterExpanding", "Cluster is expanding")
```

Comment:
I still don't think we need an expanding or shrinking message. A generic message for "configuring OSDs" seems sufficient from what you already have elsewhere. Unless there is a strong requirement to detect conditions for temporary conditions like this, I would suggest we not do it. Simple code is easier to maintain over time. Do you have more details on the requirement?

Reply:
My idea was that everytime an additional OSD is added to the cluster, it would show the `Expanding` phase and also a message saying maybe `Cluster is adding additional OSD` or a similar one. And also even if there is one OSD available, then add a condition `Available` to the cluster with a message `Storage is Available on cluster`. @travisn 

I'll try to simplify the codes in osd.go.

Reply:
I'm curious to know more about the scenario for consumers of the conditions. Have you heard a requirement where some consumer of rook needs to know through the conditions that the cluster was expanded? If someone needs to confirm the cluster was expanded, there are already ways such as looking at how many osd daemons are running or querying the ceph status in the toolbox. The event of expanding the cluster is such a short/temporary event that it doesn't seem worth raising a condition for it. Then someone will ask for conditions for expanding mons, rgw, or other daemons. Conditions just don't seem necessary to raise all those details of what the operator is doing.

Reply:
Alright. I can see your point. Do we then need the StorageAvailable?

Reply:
Agree with @travisn  . But, this could be meaningful to users who only have the UI access (in OpenShift/OLM) and not the CLI access to inspect.

Reply:
@travisn ^

Reply:
StorageAvailable is intending to represent that there is at least one OSD? That may not be sufficient as a health indication because you likely require 3 OSDs in order to have a healthy cluster. What you really want to know is if the cluster is healthy. What if we added these three conditions instead of that one:
- "HealthOK": no health warnings or errors
- "HealthWarning": the message is the first health warning indicated by ceph
- "HealthError": the message is the first health error indicated by ceph

Reply:
`HealthOK`, `HealthWarning` and `HealthError` is already shown in the ceph status right? These also has the field for last transition and even the previous Health state. Wouldn't it be a repeated information if this thing showed into the Conditions also. (Condition is just below the ceph status)

Reply:
@leseb As per the above comment made by @travisn I don't think we can make an implementation for these Conditions on the rook (at least for now). 
But I can make the messages less generic.

Reply:
Then close the PR :)

Reply:
I'll change this PR into addressing less generic messages for the conditions and push it in by tomorrow if its okay @leseb ?